examples/black_scholes_greeks.py

```diff
@@ -0,0 +1,212 @@
+    for i in range(n_vol_steps):
+        V[:, i, :] += i * vol_step
+
+    return CALL, PUT, S, K, T, R, V
+
+
+# A cumulative distribution function (CDF) describes the probabilities
+# of a random variable having values less than or equal to X
+# from Deisenroth, Marc Peter; Faisal, A. Aldo; Ong, Cheng Soon (2020).
+# Mathematics for Machine Learning
+def normCDF(d):
+    A1 = 0.31938153
+    A2 = -0.356563782
+    A3 = 1.781477937
+    A4 = -1.821255978
+    A5 = 1.330274429
+
+    K = 1.0 / (1.0 + 0.2316419 * np.absolute(d))
+
+    cnd = (
+        RSQRT2PI
+        * np.exp(-0.5 * d * d)
+        * (K * (A1 + K * (A2 + K * (A3 + K * (A4 + K * A5)))))
+    )
+
+    return np.where(d > 0, 1.0 - cnd, cnd)
+
+
+# In order to compute this efficiently, an expansion approximation
+# from John C. Hull (1997) “Options, Futures, and Other Derivatives”) is used.
+def normPDF(d):
+    return RSQRT2PI * np.exp(-0.5 * d * d)
+
+
+# Black Scholes Model estimates the theoretical value of derivatives
+# based on other investment instruments, taking into account the
+# impact of time and other risk factors.
+def black_scholes(out, S, K, R, T, V, CP, greek):
+    stdev = V * np.sqrt(T)
+    df = np.exp(-R * T)
+    d1 = (np.log(S / K) + (R + 0.5 * V * V) * T) / stdev
+    d2 = d1 - stdev
+    nd1 = normCDF(CP * d1)
+    nd2 = normCDF(CP * d2)
+
+    # the Greeks are the quantities representing the sensitivity of the
+    # price of derivatives such as options to a change in underlying
+    # parameters on which the value of an instrument or portfolio of
+    # financial instruments is dependent.
+    # from Haug, Espen Gaardner (2007). The Complete Guide to Option Pricing
+    # Formulas. McGraw-Hill Professional. ISBN 9780071389976.
+    if greek == Greeks.PREM:
+        out[...] = CP * (S * nd1 - K * df * nd2)
+    elif greek == Greeks.DELTA:
+        out[...] = CP * nd1
+    elif greek == Greeks.VEGA:
+        out[...] = S * np.sqrt(T) * normPDF(d1)
+    elif greek == Greeks.GAMMA:
+        out[...] = normPDF(d1) / (S * V * np.sqrt(T))
+    elif greek == Greeks.VANNA:
+        out[...] = -d2 * normPDF(d1) / V
+    elif greek == Greeks.VOLGA:
+        out[...] = S * np.sqrt(T) * d1 * d2 * normPDF(d1) / V
+    elif greek == Greeks.THETA:
+        out[...] = -(
+            0.5 * S * V / np.sqrt(T) * normPDF(d1) + CP * R * df * K * nd2
+        )
+    else:
+        raise RuntimeError("Wrong greek name is passed")
+
+
+def run_black_scholes_benchmark(
+    n_vol_steps, n_t_steps, n_money_steps, iters, warmup
+):
+    print("Start black_scholes")
+    CALL, PUT, S, K, T, R, V = initialize(
+        n_vol_steps, n_t_steps, n_money_steps, np.float32
+    )
+
+    timer.start()
+    for i in range(-warmup, iters):
+        if i == 0:
+            timer.start()
+        for g in Greeks:
+            black_scholes(CALL[g.value], S, K, R, T, V, 1, g)
+            black_scholes(PUT[g.value], S, K, R, T, V, -1, g)
+
+    total = (timer.stop()) / iters
```

Comment:
I think you want to report the total time (i.e. not divided by the number of iterations?) At least that's what the other tests do.

Reply:
sure